code/game/machinery/camera/presets.dm

```diff
@@ -68,17 +68,11 @@
 //This camera type automatically sets it's name to whatever the area that it's in is called.
 /obj/machinery/camera/autoname/New()
 	..()
-	spawn(10)
-		number = 1
-		var/area/A = get_area(src)
-		if(A)
-			for(var/obj/machinery/camera/autoname/C in GLOB.machines)
-				if(C == src) continue
-				var/area/CA = get_area(C)
-				if(CA.type == A.type)
-					if(C.number)
-						number = max(number, C.number+1)
-			c_tag = "[A.name] #[number]"
+	var/static/list/id_by_area = list()
+	var/area/A = get_area(src)
+	if(!id_by_area[A])
+		id_by_area[A] = 0
```

Comment:
I took this out in the example earlier in discord because this if block is unnecessary